src/parser/variable.js

```diff
@@ -43,7 +43,11 @@ module.exports = {
     ) {
       result = this.node();
       const name = this.read_namespace_name();
-      if (this.token != this.tok.T_DOUBLE_COLON && this.token != "(") {
+      if (
+        this.token != this.tok.T_DOUBLE_COLON &&
+        this.token != "(" &&
+        ["parentreference", "selfreference"].indexOf(name.kind) === -1
```

Comment:
it's micro optimisations, but as it's potentially executed many times, it's in principle faster to write this expression like this : 

```
 name.kind !== "parentreference" && 
 name.kind !== "selfreference"
```